recipes/LibriSpeech/ASR/CTC/hparams/train_hf_wav2vec_k2.yaml

```diff
@@ -0,0 +1,213 @@
+#  - A working installation of k2 (and kaldilm if you want to use ARPA LMs).
+#
+# Authors: Zeyu Zhao 2023
+#          Georgios Karakasidis 2023
+#          Pierre Champion 2023
+# ################################
+
+# Seed needs to be set at top of yaml, before objects with parameters are made
+seed: 1111
+__set_seed: !apply:torch.manual_seed [!ref <seed>]
+output_folder: !ref results/train_wav2vec2_char_k2/<seed>
+save_folder: !ref <output_folder>/save
+train_log: !ref <output_folder>/train_log.txt
+
+# URL for the biggest Fairseq english wav2vec2 model.
+wav2vec2_hub: facebook/wav2vec2-large-960h-lv60-self
+wav2vec2_folder: !ref <save_folder>/wav2vec2_checkpoint
+
+# Data files
+data_folder: !PLACEHOLDER # e,g./path/to/LibriSpeech
+# noise/ris dataset will automatically be downloaded
+# data_folder_rirs: !ref <data_folder>
+train_splits: ["train-clean-100", "train-clean-360", "train-other-500"]
+dev_splits: ["dev-clean", "dev-other"]
+test_splits: ["test-clean", "test-other"]
+skip_prep: False
+ckpt_interval_minutes: 25 # save checkpoint every N min
+train_csv: !ref <output_folder>/train.csv
+valid_csv: !ref <output_folder>/dev-clean.csv
+test_csv:
+   - !ref <output_folder>/test-clean.csv
+   - !ref <output_folder>/test-other.csv
+   - !ref <output_folder>/dev-clean.csv
+   - !ref <output_folder>/dev-other.csv
+
+# For k2 CTC training
+lang_dir: !ref <output_folder>/lang
+vocab_file: !ref <data_folder>/librispeech-vocab.txt
+sil_prob: 0.
+add_word_boundary: True
+# For k2 decoding
+test_search_beam: 32
+# Beam size (for decoding)
+test_output_beam: 8
+test_min_active_state: 300
+test_max_active_state: 3000
+# Acoustic scale (mutliplied by the log probs)
+ac_scale: 1.5
+compose_HL_with_G: False
+# 1best or whole-lattice-rescoring
+# decoding_method: whole-lattice-rescoring
+decoding_method: 1best
+# LM scale to be used for rescoring. Only used if rescoring
+rescoring_lm_scale: 0.4
+# This is where the 3gram and (optionally) 4gram LM are stored
+# They can be in either ARPA or FST format. If the former, then
+# the FST equivalent will be created in the same directory by
+# using kaldilm.
+lm_dir: lm
+# The ARPA LM files are located under the lm_dir.
+# - Use (recommended):
+#     - 3-gram_sb.arpa
+#     - 4-gram_sb.arpa
+#     To downloads speechbrain pretrained models (trained on train-960+librispeech-lm-norm.txt, 214k words)
+# - Use:
+#    - 3-gram.arpa
+#    - 3-gram.pruned.1e-7.arpa
+#    - 3-gram.pruned.3e-7.arpa
+#    - 4-gram.arpa
+#    To downloads http://www.openslr.org/resources/11/ pretrained models (trained on librispeech-lm-norm.txt, 200k words)
+# - Use another name for a model you trained yourself.
+#    If the arpa does not exist in the lm_dir, the recipe will create the
+#    necessary files for n-gram training and provide you an example on how
+#    kenlm (https://github.com/kpu/kenlm) can be run.
+G_arpa: 3-gram_sb.arpa
+G_rescoring_arpa: 4-gram_sb.arpa
+# caching: False
```

Comment:
why is it not being used ?

Reply:
I don't understand the question.
It is used train_with_wav2vec_k2.py and `sbk2.lattice_decoder.get_decoding` when `compose_HL_with_G` is set to `true` or `decoding_method` is set to `whole-lattice-rescoring`